src/Sdk/DTWebApi/WebApi/Runner.cs

```diff
@@ -1,18 +1,73 @@
+using System;
 using Newtonsoft.Json;
 
 namespace GitHub.DistributedTask.WebApi
 {
     public class Runner
     {
-        /// <summary>
-        /// Name of the agent
-        /// </summary>
+
+        public class Authorization
+        {
+            /// <summary>
+            /// The url to refresh tokens
+            /// </summary> 
+            [JsonProperty("authorization_url")]
+            public Uri AuthorizationUrl
+            {
+                get;
+                internal set;
+            }
+
+            /// <summary>
+            /// The url to connect to to poll for messages
+            /// </summary> 
+            [JsonProperty("server_url")]
+            public string ServerUrl
+            {
+                get;
+                internal set;
+            }
+
+            /// <summary>
+            /// The client id to use when connecting to the authorization_url
+            /// </summary>
+            [JsonProperty("client_id")]
+            public string ClientId
+            {
+                get;
+                internal set;
+            }
+        }
+
         [JsonProperty("name")]
         public string Name
         {
             get;
             internal set;
         }
 
+        [JsonProperty("id")]
+        public Int32 Id
+        {
+            get;
+            internal set;
+        }
+
+        [JsonProperty("authorization")]
+        public Authorization RunnerAuthorization
+        {
+            get;
+            internal set;
+        }
+
+        public void ApplyToTaskAgent(TaskAgent agent)
```

Comment:
This feels like it might be blurring encapsulation lines.  Maybe it's better as a static method in a static class?  Or maybe it's better as  a method on `TaskAgent`?

Reply:
great point. Updated.

Reply:
I opted to just pull this out of the method and make it explicit at the call site rather than burying it